vignettes/pandoc_table.Rmd

```diff
@@ -0,0 +1,195 @@
+
+```{r}
+m <- data.frame('Value\n1', 'Value\n2')
+colnames(m) <- c('Multiline\nCol1', 'Multiline\nCol2')
+pandoc.table(m, keep.line.breaks = TRUE)
+m <- mtcars[1:3, 1:4]
+pandoc.table(m)
+```
+
+### simple tables
+
+`simple` tables are have more compact syntax that all other styles, but don't they support multiline cells:
+
+```{r, error=TRUE}
+m <- mtcars[1:3, 1:4]
+pandoc.table(m, style = 'simple')
+m <- data.frame('Value\n1', 'Value\n2')
+colnames(m) <- c('Multiline\nCol1', 'Multiline\nCol2')
+pandoc.table(m, keep.line.breaks = TRUE, style='simple')
+```
+
+### grid tables
+
+`grid` format is really handy for `emacs` users ([Emacs table mod](http://table.sourceforge.net/)) and it does support block elements (multiple paragraphs, code blocks, lists, etc.) inside cells. Alignments are not supported by most parsers, meaning that even though `pander` will produce a table with alightnment, it will be lost during conversion from `markdown` to `HTML/PDF/DOCX`, and cells can't span multiple columns or rows.
+
+```{r, error=TRUE}
+m <- mtcars[1:3, 1:4]
+pandoc.table(m, style = 'grid')
+m <- data.frame('Value\n1', 'Value\n2')
+colnames(m) <- c('Multiline\nCol1', 'Multiline\nCol2')
+pandoc.table(m, keep.line.breaks = TRUE, style='grid')
+```
+
+### rmarkdown tables
+
+`rmarkdown` or pipe table format is often used directly with `knitr`, since it was supporters by the first versions of the `markdown` package. The beginning and ending pipe characters are optional, but pipes are required between all columns:
+
+```{r, error=TRUE}
+m <- mtcars[1:3, 1:4]
+pandoc.table(m, style = 'rmarkdown')
+m <- data.frame('Value\n1', 'Value\n2')
+colnames(m) <- c('Multiline\nCol1', 'Multiline\nCol2')
+pandoc.table(m, keep.line.breaks = TRUE, style='rmarkdown')
+```
+
+## Cell alighnment
+
+`pander` allows users to control cell alignment (`left`, `right` or `center/centre`) in a table directly by setting the `justify` parameter when calling `pander/pandoc.table`. Note that it is possible to specify alighnment for each column separately by supplying a vector to justify:
+
+```{r}
+pandoc.table(head(iris[,1:3], 2), justify = 'right')
+pandoc.table(head(iris[,1:3], 2), justify = c('right', 'center', 'left'))
+```
+
+Another possible way to define alignment is using a permanent for `table.alignment.default/table.alignment.rownames` in `panderOptions` (prefered way) or using `set.alignment` function which support setting alignment separately for cells and rownames:
+
+```{r}
+set.alignment('left', row.names = 'right') # set only for next table since permanent parameter is falce
+pandoc.table(mtcars[1:2,  1:5])
+```
+
+Interesting application for this functionality is specifying a function that takes the R object as its argument to compute some unique alignment for your table based on e.g. column values or variable types:
```

Comment:
Can we add a line here on what is `panderOption`? I think it might be a bit confusing to the 1st time reader why we used `set.alignment` previously, and what's that new function. We could mention that there are many other *global options* as well, and linking to http://rapporter.github.io/pander/#general-options